### PR TITLE
support download from http link

### PR DESCRIPTION
**Pull request by runzexia**
Signed-off-by: runzexia <runzexia@yunify.com>

### CODE REVIEW

**Comment by magicsong** [comment]
is it possible to use `runtimeImage` only as it seems unneccessary to 'build' a binary?

File: pkg/scm/scm.go [hunk 1]
```diff
@@ -16,13 +17,15 @@ var glog = utilglog.StderrLog
...
 // the sources from the repository.
-func DownloaderForSource(fs fs.FileSystem, s *git.URL, forceCopy bool) (build.Downloader, error) {
+func DownloaderForSource(fs fs.FileSystem, s *git.URL, forceCopy bool, isBinaryDownload bool) (build.Downloader, error) {
 	glog.V(9).Infof("DownloadForSource %s", s)
 
 	if s == nil {
 		return &empty.Noop{}, nil
 	}
-
+	if isBinaryDownload {
```
[the remark this answers is not in the document]
Comment (runzexia) [reply pkg/scm/scm.go]: good point

**Comment by runzexia** [comment]
> is it possible to use `runtimeImage` only as it seems unneccessary to 'build' a binary?

In terms of image size, it is similar to a runtimeImage, but it will actually be a builderImage because it requires a build script, although it may be very simple

**Pull request merged**